src/components/PostDetails.tsx

```diff
@@ -1,117 +1,129 @@
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import { CommentsList } from './CommentsList';
+import { deleteComment, getCommentsToPost } from '../api/mate';
+import { Comment } from '../types/Comment';
+
+interface Props {
+  selectedPost: Post;
+}
+
+export const PostDetails: React.FC<Props> = ({ selectedPost }) => {
+  const {
+    id,
+    title,
+    body,
+  } = selectedPost;
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [showError, setShowError] = useState(false);
+  const [showNewCommentForm, setShowNewCommentForm] = useState(false);
+
+  const getCommentsFromServer = useCallback(async (postId: number) => {
+    setIsLoading(true);
+    setShowError(false);
+    try {
+      const result = await getCommentsToPost(postId);
+
+      setComments(result);
+    } catch {
+      setShowError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  }, []);
+
+  const addComment = (comment: Comment) => {
+    setComments(prev => [
+      ...prev,
+      comment,
+    ]);
+  };
+
+  const removeCommentById = async (commentId: number) => {
+    const commentToDelete = comments.find(comment => comment.id === commentId);
+    const index = commentToDelete ? comments.indexOf(commentToDelete) : -1;
```

Comment:
`comments.find` will return you an object. And `indexOf` works only for Primitive data types (string, number, boolean and so on). So the `comments.indexOf(commentToDelete)` will always return `-1`, why do you need a ternary operator here? When there is no value found you get -1 by default...